hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -403,6 +360,34 @@ protected ByteString readChunk(final ChunkInfo chunkInfo,
     return xceiverClient.getPipeline().getNodes();
   }
 
+  private CheckedFunction<ContainerProtos.ContainerCommandResponseProto, IOException>
+      validator = (response) -> {
+    ReadChunkResponseProto readChunkResponse;
+    final ChunkInfo chunkInfo = chunks.get(chunkIndex);
```

Comment:
The chunkIndex might get updated between the readChunk call and the validator call. 
Also, with HDDS-1496, we are going to support partial chunk reads. So we would have to somehow let the validator know the chunkIndex, offset and length which was used in the readChunk call.

Reply:
Addressed in the latest patch.